app/src/common/shared/org/mozilla/vrbrowser/browser/components/GeckoWebExtensionRuntime.kt

```diff
@@ -0,0 +1,281 @@
+            webExtensionDelegate: WebExtensionDelegate
+    ) {
+        this.webExtensionDelegate = webExtensionDelegate
+
+        val promptDelegate = object : WebExtensionController.PromptDelegate {
+            override fun onInstallPrompt(ext: org.mozilla.geckoview.WebExtension): GeckoResult<AllowOrDeny>? {
+                val extension = GeckoWebExtension(ext, runtime)
+                return if (webExtensionDelegate.onInstallPermissionRequest(extension)) {
+                    GeckoResult.ALLOW
+                } else {
+                    GeckoResult.DENY
+                }
+            }
+
+            override fun onUpdatePrompt(
+                    current: org.mozilla.geckoview.WebExtension,
+                    updated: org.mozilla.geckoview.WebExtension,
+                    newPermissions: Array<out String>,
+                    newOrigins: Array<out String>
+            ): GeckoResult<AllowOrDeny>? {
+                // NB: We don't have a user flow for handling updated origins so we ignore them for now.
+                val result = GeckoResult<AllowOrDeny>()
+                webExtensionDelegate.onUpdatePermissionRequest(
+                        GeckoWebExtension(current, runtime),
+                        GeckoWebExtension(updated, runtime),
+                        newPermissions.toList()
+                ) {
+                    allow -> if (allow) result.complete(AllowOrDeny.ALLOW) else result.complete(AllowOrDeny.DENY)
+                }
+                return result
+            }
+        }
+
+        val debuggerDelegate = object : WebExtensionController.DebuggerDelegate {
+            override fun onExtensionListUpdated() {
+                webExtensionDelegate.onExtensionListUpdated()
+            }
+        }
+
+        runtime.webExtensionController.promptDelegate = promptDelegate
+        runtime.webExtensionController.setDebuggerDelegate(debuggerDelegate)
+    }
+
+    /**
+     * See [Engine.listInstalledWebExtensions].
+     */
+    override fun listInstalledWebExtensions(onSuccess: (List<WebExtension>) -> Unit, onError: (Throwable) -> Unit) {
+        runtime.webExtensionController.list().then({
+            val extensions = it?.map {
+                extension ->
+                GeckoWebExtension(extension, runtime)
+            } ?: emptyList()
+
+            val privateBrowsingMigrated = sharedPref.getBoolean(MOZAC_HAS_RESET_WEBEXT_PRIVATE_BROWSING_DEFAULT, false)
+            extensions.forEach { extension ->
+                // As a workaround for https://bugzilla.mozilla.org/show_bug.cgi?id=1621385,
+                // we set all installed extensions to be allowed in private browsing mode.
+                // We need to revert back to false which is now the default.
+                if (!extension.isBuiltIn() && extension.isAllowedInPrivateBrowsing() && !privateBrowsingMigrated) {
+                    setAllowedInPrivateBrowsing(extension, false)
+                }
+
```

Comment:
Do we need this? Looks like some compat code that won't affect us.

Reply:
Not really. Removed.